src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -718,6 +718,19 @@ class SemiStructuredAccessorSegment(BaseSegment):
     )
 
 
+class TimeZoneGrammar(BaseSegment):
+    """Casting to Time Zone.
+
+    BigQuery uses Double Quotes
+    https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timezone_definitions
+    """
+
+    type = "time_zone_grammar"
+    match_grammar = AnyNumberOf(
+        Sequence("AT", "TIME", "ZONE", Ref("DoubleQuotedLiteralSegment")),
```

Comment:
In my use-case the timezone will come from a column in the table itself. I think this case is not covered here, right?

Reply:
I didn't know that was possible and thought your example was wrong. Will add that support.

Reply:
Added in https://github.com/sqlfluff/sqlfluff/pull/3004/commits/b5555a38c190dcd07ac056f8d9ac53294ed1b91a